Code/Yospace/Source/YospaceManager.swift

```diff
@@ -0,0 +1,151 @@
+        self.yospaceSession = nil
+        self.source = nil
+        self.id3MetadataHandler = nil
+        self.playerEventsHandler = nil
+        self.yospaceNotificationsHandler = nil
+    }
+
+    func destroy() {
+        self.reset()
+        self.adIntegrationHandler = nil
+        self.adIntegrationController = nil
+        self.eventDispatcher.clear()
+    }
+
+    private func onSessionCreate(session: YOSession) {
+        let state: YOSessionState = session.sessionState
+        if state == .initialised || state == .noAnalytics {
+            self.setupManager(session: session)
+        } else if state == .failed {
+            let message: String = YOSession.YOSessionResultCode(rawValue: session.resultCode)?.message() ?? "Yospace: Session could not be initialised"
+            let error = YospaceError.error(msg: message)
+            self.adIntegrationController?.fatalError(error: error, code: .AD_ERROR)
+        }
+    }
+
+    private func setupManager(session: YOSession) {
+        if let playbackUrlStr: String = session.playbackUrl,
+           let playbackUrl: URL = .init(string: playbackUrlStr) {
+            self.yospaceSession = session
+            (self.adIntegrationHandler as? YospaceHandler)?.session = session
+            self.id3MetadataHandler = YospaceID3MetadataHandler(player: self.player, session: session)
+            self.playerEventsHandler = THEOplayerEventsHandler(player: self.player, session: session)
+            if let controller: THEOplayerSDK.ServerSideAdIntegrationController = self.adIntegrationController {
+                self.yospaceNotificationsHandler = YospaceNotificationsHandler(session: session, adIntegrationController: controller, player: self.player)
+            }
+            if let source: YospaceManagerSource = self.source {
+                let typedSource: THEOplayerSDK.TypedSource = source.1
+                typedSource.src = playbackUrl
+                let sourceDescription: THEOplayerSDK.SourceDescription = source.0
+                self.isSettingSource = true
+                self.player.source = sourceDescription
+                self.isSettingSource = false
+            }
+            self.yospaceSession?.setPlaybackPolicyHandler(DefaultPlaybackPolicy(playbackMode: session.playbackMode))
+            let event: SessionAvailableEvent = .init(date: Date())
+            self.eventDispatcher.dispatchEvent(event: event)
+        }
+    }
+
+    deinit {
+        self.destroy()
+    }
+}
+
+class YospaceHandler: THEOplayerSDK.ServerSideAdIntegrationHandler {
+    weak var player: THEOplayerSDK.THEOplayer?
+    weak var manager: YospaceManager?
+    var session: YOSession?
+
+    init(player: THEOplayerSDK.THEOplayer, manager: YospaceManager) {
+        self.player = player
+        self.manager = manager
+    }
+
+    func setSource(source: SourceDescription) -> Bool {
+        self.manager?.createYospaceSource(sourceDescription: source, sessionProperties: nil)
+        return true
```

Comment:
Why are you _always_ returning `true` here?

If `source` is *not* a Yospace source, then `createYospaceSource()` won't do anything. But because you're returning `true` here, the player *also* won't do anything with the source, since it assumes that the ad integration handler took responsibility of the call. So the end result is that the source... never gets set on the player at all?

Reply:
The connector returns `true` to signal feedback to the core SDK that it will resolve the source. This will trigger the core to await a new source set. This is done as a workaround to the concurrency limitation.

What is missing here, as you pointed out, is handling the scenario where the source resolution fails. I will make changes to handle the unhappy path. After that the connector will either:
1. Successfully resolve the Yospace source and set a new source on the player
2. Fail and throw a player error

Reply:
Done, propagates the error with `ServerSideAdIntegrationController.fatalError`.